ta/pkcs11/src/processing_aes.c

```diff
@@ -332,12 +332,7 @@ enum pkcs11_rc tee_ae_decrypt_final(struct pkcs11_session *session,
 		res = TEE_AEDecryptFinal(session->processing->tee_op_handle,
 					 NULL, 0, data_ptr, &data_size,
 					 ctx->pending_tag, ctx->tag_byte_len);
-
-		if (!data_size) {
-			TEE_Free(data_ptr);
-			data_ptr = NULL;
-			DMSG("Is this expected from the Core API?");
-		}
+		assert(!res && data_size);
```

Comment:
Asserting `!res` may be a bit much, runtime errors can happen. However `data_size` should never be set to zero on output so `assert(data_size)` does make sense (but may not be that useful).

Reply:
Ok, i'll fix that. Thanks.